DiscordBot.DataAccess/EventsSheetsService.cs

```diff
@@ -208,15 +207,21 @@ public async Task<IEnumerable<DiscordEvent>> ListEventsAsync()
             }
         }
 
-        private static ServiceAccountCredential GetCredential(string path = "credentials.json")
+        private static ServiceAccountCredential GetCredential()
         {
-            using var stream =
-                new FileStream(path, FileMode.Open, FileAccess.Read);
-
-            return GoogleCredential.FromStream(stream)
-                       .CreateScoped(scopes)
-                       .UnderlyingCredential as ServiceAccountCredential
-                   ?? throw new EventsSheetsInitialisationException("Credential maker returned null");
+            var clientEmail = Environment.GetEnvironmentVariable("GOOGLE_CLIENT_EMAIL");
+            var projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+            var privateKeyId = Environment.GetEnvironmentVariable("GOOGLE_PRIVATE_KEY_ID");
+            var privateKey = Environment.GetEnvironmentVariable("GOOGLE_PRIVATE_KEY");
+            return new ServiceAccountCredential(
+                new ServiceAccountCredential.Initializer(clientEmail)
```

Comment:
This is a bit hard to read. I'd recommend creating the initialiser first then the credential 🙂

Reply:
Done in 4d9b757 🙂